Restructure build for individual pathogenesis manuscript

agitter opened the pull request:
<!-- Hi there! Please use the template below as a guide for opening a pull request.
This template is designed to help the pull request process go smoother.
If anything doesn't make sense to you, please open the pull request anyway and leave entries blank as needed.
Thank you for your contribution!! -->

### Description of the proposed additions or changes
<!-- If the additions or changes aren't self explanatory feel free to leave a description or notes here. -->
This pull request will experiment with build script modifications.  The goal is to build a DOCX of only the pathogenesis content.

### Related issues
<!-- GitHub will link issues to pull requests if you list the related issue numbers like "#2, #17". -->
#732 #762

### Suggested reviewers (optional)
<!-- If there are particular people you think should review this pull request, please list their GitHub handle like "@rando2".-->
@rando2

### Checklist
<!-- Please mark the following tasks as complete by putting an "x" in the box like "[x]".
If a particular item is not applicable to your pull request, please delete the item. -->
- [x] Text is formatted so that each sentence is on its own line.
<!-- see why this is recommended here:https://github.com/greenelab/covid19-review/blob/627a5f45a2d1fcb2150d1466df63bb848f22a3d4/USAGE.md#manubot-markdown -->
- [x] Pre-prints cited in this pull request have a GitHub issue opened so that they can be reviewed.

agitter commented:
The basic structure for building a docx of the pathogenesis manuscript is working locally.  I disabled AppVeyor builds temporarily.  Here is the docx generated by GitHub Actions: [pathogenesis-manuscript.docx](https://github.com/greenelab/covid19-review/files/5818721/pathogenesis-manuscript.docx)

I haven't tested anything with the author list or consortium members for the pathogenesis manuscript.  I'll clean up the broken svg images in the author list when I work on that.

rando2 commented:
This looks fantastic, @agitter. I'm actually going to email out the docx to everyone so they have a almost-final version! Would it be helpful if I started updating the metadata.yaml to reflect who is on this author list?

agitter commented:
Let's wait until I can test a couple ideas for how to add the paper-specific author information to the metadata file.  Hopefully I can work on it more later today.

rando2 commented:
That sounds great @agitter! I'm happy to do whatever I can to help -- thank you so much for everything you've already done on this, it looks awesome and will really help with running things by people!

rando2 commented:
Hi @agitter, I am totally prepared to add in the roles for each manuscript as we finalize them on each manuscript hub issue (e.g., #744). Honestly this will be a great way for me to keep track of everything while going through the commit histories!

I think @cgreene had suggested using a random element to this one like in the Deep Review (so people in category 6 are randomized)-- would that be hard to switch out?

I think for the merged HTML document, intuitively I would say to just use the set (in the python sense of de-duplicated collection) of contributions. 

I think that addresses your questions, I might try opening a PR later today to test my ability to assign per-document author contributions!

cgreene commented:
I think I was probably overly complicating things with the randomization. I'm happy to drop that suggestion in favor of simplicity + marking contribution categories.

agitter commented:
Now that I'm setting the author order in a Python script anyway, randomization isn't necessarily too complicated.  Randomly re-ordering every build like deep review v2 would be fairly easy.  Randomly ordering once and then locking it would take more work, but I won't rule it out of that's what you (@rando2 and @cgreene) think is fair to the authors.

@rando2 I'll make a few more changes to support the individual manuscript contributions, and then this may be ready to review.  Because I edited the metadata file heavily, I'd like to merge this before you make any changes to it.

Taking the union of contributions should be possible.  I'll play around with that.

agitter commented:
@rando2 I added support for setting the contributions in the merged manuscript to be the union of contributions from the individual manuscripts.  If contributions are not set for an author, they will be stored as **MISSING** and print a warning in the build log.  I added example contributions for you in the metadata to demonstrate the behavior.

If this build passes, it could be a good time to add some usage documentation and request a review.  We can work to merge this so you can start adding the real metadata for the pathogenesis manuscript.  I could work on author ordering more or listing the consortium members in a separate pull request.

agitter commented:
This is ready for review.  I added some documentation and support for the consortium author.  Now there is an author "COVID-19 Review Consortium" in the metadata file.  This author should be listed and ordered in each individual manuscript.  All authors except COVID-19 Review Consortium and the Immunology Institute of the Icahn School of Medicine are listed under the author list as consortium members.  Feel free to modify the consortium name, how the members are listed in the front matter, etc.

I also modified contributions in the merged manuscript so that authors who are not yet listed as authors of any individual manuscript can use their existing general contributions.

rando2 commented on build/README.md:
```diff
@@ -16,6 +16,11 @@ Format conversion is done using [Pandoc](https://pandoc.org/MANUAL.html).
 Each file specifies a set of pandoc `--defaults` options for a given format.
 To change the options, either edit the YAML files directly or add additional `--defaults` files.
 
+If the `BUILD_HTML` or `BUILD_PDF` environment variables are not false or the `BUILD_DOCX` environment variable is set to `true`, the build script will produce a merged manuscript
+with content from all individual manuscripts.
```
```suggestion
If the `BUILD_HTML` or `BUILD_PDF` environment variables are not false or the `BUILD_DOCX` environment variable is set to `true`, the build script will produce a merged manuscript with content from all individual manuscripts.
```

rando2 reviewed (approved):
Amazing Tony! I'll merge this in and then go back and update the metadata for the authors, if that sounds good (to reflect the author order for pathogenesis and to move anyone who is not on pathogenesis to 1 paper that is a good fit for them!)
